### PR TITLE
Add MetaMask SwitchChain button

### PR DESCRIPTION
**Pull request by en0c-026**
added SwitchChain component
if the network is not added to Metamask, the network is added first and then changed.
Added dependency on @ethers-project/bytes to make use of the hexValue function, to parse the chainId into a hex value.

Related [bounty](https://gitcoin.co/issue/29146)

### CODE REVIEW

File: src/sections/Section4.jsx [hunk 1]
```diff
@@ -39,6 +62,43 @@ function RpcInformation(){
...
+          params: [{ chainId: chainIdHex }]
+        })
+      } catch (error) {
+        try {
+          if (error.code === 4902) {
+            await rawEthereumProvider.request({
+              method: 'wallet_addEthereumChain',
+              params: [{
+                chainId: chainIdHex,
+                chainName: name,
```
Comment (mjadach-iv) [inline src/sections/Section4.jsx:85]: Please use exactly the same data that you can see under the RPC dropdown when switching.

**Review by mjadach-iv — CHANGES_REQUESTED** [review]
After those small changes, PR should be good to go.

File: src/sections/Section4.jsx [hunk 2]
```diff
@@ -19,6 +20,28 @@ const SSection = styled(Section)`
...
+`
+
+function SwitchChain(props){
+    return (
+      <SwitchContainer>
+          <SwitchButton
+            onClick={props.handleSwitchChain}
+          >
+              Switch
+          </SwitchButton>
```
[the remark this answers is not in the document]
Comment (en0c-026) [reply src/sections/Section4.jsx]: The suggested change was applied

File: src/sections/Section4.jsx (hunk 1 above)
Comment (en0c-026) [reply src/sections/Section4.jsx]: The same network data from the current state of the component is now used to add the new network.
There was a small bug with the tab size when formatting the document in Visual Studio, the original tab size was restored.

**Review by mjadach-iv — CHANGES_REQUESTED** [review]
Looks good. Only needs to be free of conflicts in order to merge.

**Review by mjadach-iv — APPROVED** [review]
LGTM

**Comment by nionis** [comment]
Bounty paid, thanks @en0c-026 .

**Comment by en0c-026** [comment]
Thank you!

**Pull request merged**